src/Microsoft.ML.OnnxConverter/OnnxContextImpl.cs

```diff
@@ -279,13 +279,27 @@ public override List<long> RetrieveShapeOrNull(string variableName)
         }
 
         /// Adds constant tensor into the graph.
+        public override string AddInitializer(bool value, string name = null, bool makeUniqueName = true)
+        {
+            name = AddVariable(name ?? "bool", makeUniqueName);
+            _initializers.Add(OnnxUtils.MakeInt32(name, typeof(bool), value ? 1 : 0));
```

Comment:
Why do booleans get converted to ints?

Reply:
Please see OnnxML.cs line 2868. According to the protobuf definition, boolean values are stored in the int32Data_ field. 

---
In reply to: [377273444](https://github.com/dotnet/machinelearning/pull/4815#discussion_r377273444) [](ancestors = 377273444)